pkg/apis/extensions/types.go

```diff
@@ -208,7 +208,7 @@ type DeploymentSpec struct {
 	Selector map[string]string `json:"selector,omitempty"`
 
 	// Template describes the pods that will be created.
-	Template *api.PodTemplateSpec `json:"template,omitempty"`
+	Template api.PodTemplateSpec `json:"template"`
```

Comment:
Have we given up on ever having a `TemplateRef`?  As I recall, that's why this is a pointer for `ReplicationControllerSpec` and I would expect congruence here.